shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/ProxyBackendHandlerFactory.java

```diff
@@ -160,6 +159,16 @@ public static ProxyBackendHandler newInstance(final DatabaseType databaseType, f
         backendHandler = DatabaseAdminBackendHandlerFactory.newInstance(databaseType, sqlStatementContext, connectionSession);
         return backendHandler.orElseGet(() -> DatabaseBackendHandlerFactory.newInstance(logicSQL, connectionSession, preferPreparedStatement));
     }
+
+    private static void checkUnsupportedDistSQLStatementInTransaction(final SQLStatement sqlStatement, final ConnectionSession connectionSession) {
+        if (connectionSession.getTransactionStatus().isInTransaction() && !isSupportedDistSQLStatement(sqlStatement)) {
+            throw new UnsupportedOperationException("Non-query dist sql is not supported within a transaction");
+        }
+    }
+
+    private static boolean isSupportedDistSQLStatement(final SQLStatement sqlStatement) {
```

Comment:
Could `isSupportedDistSQLStatementInTransaction` be better?

Reply:
Your are right.